ui.apps/src/main/content/jcr_root/apps/guides-components/components/guidessidenavigation/clientlibs/site/js/guidessidenavigation.ts

```diff
@@ -130,6 +142,28 @@ class GuidesNavigation {
             }
             ul.appendChild(listItem)
         }
+        const hasMore = end !== children.length;
+        if(hasMore) {
+            const button = this.getLoadMoreButton()
+            const marginDepth = 1.25*incrementer;
+            button.style.marginLeft = `${marginDepth}rem`;
+            button.addEventListener("click", () => {
+                this.doRenderLevel(ul, end, currIdx, children, parentPath, idx, incrementer, button)
+            })
+            ul.appendChild(button)
+        }
+        return ul;
+    }
+
+    renderLevel(children: Array<any>, idx: number, parentPath: string, incrementer: number, offset: number) {
```

Comment:
Why are we using <any> type instead of a defined type here?

Reply:
where exactly?

Reply:
@smongaAdb renderLevel(children: Array<any>, ...

Is there any issue with having a defined type for children Array.

Reply:
there should be a type defined, yes. Will do it.